### PR TITLE
Installation of Argo CD as part of the development setup

### PR DESCRIPTION
**Pull request by fgiloux**
This documents and automates the installation of Argo CD in the development environment.
This is a first step towards getting Pipelines Service entirely deployed and managed through GitOps.

### CODE REVIEW

File: local/kind/setup.sh [hunk 1]
```diff
@@ -69,6 +69,13 @@ for cluster in "${CLUSTERS[@]}"; do
         KIND_EXPERIMENTAL_PROVIDER=${KIND_EXPERIMENTAL_PROVIDER:-} ${KIND_CMD} create cluster \
             --config "${TMP_DIR}/${cluster}.config" \
             --kubeconfig "${TMP_DIR}/${cluster}.kubeconfig"
+
+	if [[  ${KIND_CMD} == "sudo kind" ]]; then
+		sudo chmod +r "${TMP_DIR}/${cluster}.kubeconfig"
+	fi
+
+	printf "Provisioning ingress router in ${cluster}\n"
```
Comment (Roming22) [inline local/kind/setup.sh:77]: Why use of `printf` over `echo` for simple strings?

File: local/argocd/setup.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	i=$((i+1))
+	if [ $i -gt 60 ]; then
+		printf "Ingress deployment not ready in time\n"
+		exit 1
+	fi
+done
+kubectl --kubeconfig ${KUBECONFIG} apply -f ingress.yaml
+
+printf "Argo CD installed\n"
+
```
Comment (Roming22) [inline local/argocd/setup.sh:42]: Could you also log the user in with `argocd --config "${TMP_DIR}/argocd.yaml"`?
That will create a necessary file for running `deploy.sh`.

File: local/kind/setup.sh [hunk 3]
```diff
@@ -79,4 +86,9 @@ for cluster in "${CLUSTERS[@]}"; do
 
 done
 
+NO_ARGOCD="${NO_ARGOCD:-}"
+if [[ "${NO_ARGOCD,,}" != "true" ]]; then
+	KUBECONFIG="${TMP_DIR}/${CLUSTERS[0]}.kubeconfig" ../argocd/setup.sh
+fi
+
```
Comment (Roming22) [inline local/kind/setup.sh:93]: Running `deploy.sh` requires the presence of `plnsvc.yaml` which is the kubeconfig on which to install the pipeline service.
Would you mind adding `ln -sf "${TMP_DIR}/${CLUSTERS[0]}.kubeconfig" "${TMP_DIR}/plnsvc.yaml"`?

File: local/kind/setup.sh (hunk 1 above)
Comment (fgiloux) [reply local/kind/setup.sh:77]: [more standardized/consistent](. https://unix.stackexchange.com/questions/65803/why-is-printf-better-than-echo). I try to favor printf over echo.

File: local/argocd/setup.sh (hunk 2 above)
Comment (fgiloux) [reply local/argocd/setup.sh:42]: I am currently looking at this part. My current thinking is that it does not belong here. This should be limited to argoCD installation.

File: local/kind/setup.sh (hunk 3 above)
Comment (fgiloux) [reply local/kind/setup.sh:93]: With the phase 1 approach the Pipelines service operators need to be installed on the workload clusters. There may be more than 1. I am looking at this as well. The integration with next steps can come in a separate PR.

**Comment by Roming22** [comment]
Integration with deploy.sh will be handled as a separate PR.

**Pull request merged**